src/main/java/org/jenkins/plugins/lockableresources/LockStepExecution.java

```diff
@@ -30,51 +31,62 @@
 
 	@Override
 	public boolean start() throws Exception {
-		if (LockableResourcesManager.get().createResource(step.resource)) {
-			listener.getLogger().println("Resource [" + step.resource + "] did not exist. Created.");
+		if (step.label != null && !step.label.isEmpty() && step.resource !=  null && !step.resource.isEmpty()) {
+			throw new Exception("Label and resource name cannot be specified simultaneously.");
 		}
-		listener.getLogger().println("Trying to acquire lock on [" + step.resource + "]");
-		LockableResourcesStruct resourceHolder = new LockableResourcesStruct(step.resource);
-		if(!LockableResourcesManager.get().lock(resourceHolder.required, run, getContext(), step.inversePrecedence)) {
-			// we have to wait
-			listener.getLogger().println("[" + step.resource + "] is locked, waiting...");
-		} // proceed is called inside lock otherwise
+		
+		// create resoure only when specified explicitly
+		if (step.label == null && LockableResourcesManager.get().createResource(step.resource)) {
```

Comment:
since `step.label == null` implies `step.resources != null`, it would simplify the logic to move this in the existing block below.